Engine/DataFeeds/HistoryProviderManager.cs

```diff
@@ -0,0 +1,138 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NodaTime;
+using QuantConnect.Configuration;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Lean.Engine.DataFeeds.Enumerators;
+using QuantConnect.Lean.Engine.HistoricalData;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using HistoryRequest = QuantConnect.Data.HistoryRequest;
+
+namespace QuantConnect.Lean.Engine.DataFeeds
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IHistoryProvider"/> which
+    /// acts as a wrapper to use multiple history providers together
+    /// </summary>
+    public class HistoryProviderManager : SynchronizingHistoryProvider
+    {
+        /// <summary>
+        /// Collection of history providers being used
+        /// </summary>
+        /// <remarks>Protected for testing purposes</remarks>
+        protected List<IHistoryProvider> HistoryProviders { get; } = new();
```

Comment:
- privates should go at the top
- let's move `HistoryProviderManager` under `HistoricalData` instead of `DataFeeds`